app/controllers/api/base_controller.rb

```diff
@@ -22,6 +22,11 @@ class BaseController < ActionController::Base
       render_error 'standard_error', :status => :internal_server_error, :locals => { :exception => error }
     end
 
+    rescue_from NoMethodError do |error|
+      Foreman::Logging.exception("Action failed", error)
+      render_error 'custom_error', :status => :internal_server_error, :locals => { :message => _('500 Internal Server Error') }
```

Comment:
I think it would be better if we displayed a better error message to user instead of some cryptic internal message. If instead of "500 Internal Server Error" the message was something like "Internal Server Error: the server was unable to finish the request, this might be caused by unavailability of some required service, incorrect API call or a bug on server side. Please find more information in server's production.log." then I'd definitely see the value.